packages/assets-controllers/src/TokensController.ts

```diff
@@ -960,6 +961,22 @@ export class TokensController extends BaseController<
       );
     }
   }
+
+  initApprovals() {
+    const { suggestedAssets } = this.state;
+    return Promise.all(
+      suggestedAssets
+        .filter((asset) => asset.status === SuggestedAssetStatus.pending)
+        .map((asset) =>
+          this._requestApproval(
+            asset as SuggestedAssetMeta & {
+              interactingAddress: string;
+            },
```

Comment:
This repetition of `SuggestedAssetMeta & { interactingAddress: string; }` all over the place will be no longer necessary with the new approval promise refactor.